skimage/draw/_random_shapes.py

```diff
@@ -171,46 +171,50 @@ def _generate_triangle_mask(point, image, shape, random):
 SHAPE_CHOICES = list(SHAPE_GENERATORS.values())
 
 
-def _generate_random_color(gray, min_pixel_intensity, random):
+def _generate_random_color(num_channels, intensity_range, random):
     """Generates a random color array.
 
     Parameters
     ----------
-    gray : bool
-        If `True`, the color will be a scalar, else a 3-element array.
-    min_pixel_intensity : [0-255] int
-        The lower bound for the pixel values.
+    num_channels : int
+        Number of elements representing color.
+    intensity_range : {tuple of tuples of ints, tuple of ints}, optional
+        The range of values to sample pixel values from. For grayscale images
+        the format is (min_, max_). For multichannel - ((min_, max_),) if the
+        ranges are equal across the channels, and
+        ((min_0, max_0), ... (min_N, max_N)) if they differ.
     random : np.random.RandomState
         The random state to use for random sampling.
 
     Raises
     ------
     ValueError
-        When the min_pixel_intensity is not in the interval [0, 255].
+        When the `intensity_range` is not in the interval (0, 255).
 
     Returns
     -------
     color : scalar or array
-        If gray is True, a single random scalar in the range of
-        [min_pixel_intensity, 255], else an array of three elements, each in
-        the range of [min_pixel_intensity, 255].
+        If `num_channels` is 1, a single random scalar from the
+        `intensity_range`, else an array of `num_channels` elements,
+        each from the corresponding interval of `intensity_range`.
 
     """
-    if not (0 <= min_pixel_intensity <= 255):
-        raise ValueError('Minimum intensity must be in interval [0, 255]')
-    if gray:
-        return random.randint(min_pixel_intensity, 256)
-    return random.randint(min_pixel_intensity, 256, size=3)
+    if num_channels == 1:
+        return random.randint(*intensity_range)
+    else:
+        if len(intensity_range) == 1:
+            intensity_range = intensity_range * num_channels
+        return [random.randint(r[0], r[1]+1) for r in intensity_range]
 
 
 def random_shapes(image_shape,
                   max_shapes,
                   min_shapes=1,
                   min_size=2,
                   max_size=None,
-                  gray=False,
+                  num_channels=3,
                   shape=None,
-                  min_pixel_intensity=0,
+                  intensity_range=None,
                   allow_overlap=False,
                   num_trials=100,
                   random_seed=None):
```

Comment:
I don't like `num_channels` as the API. It doesn't jive with the rest of the library. I do like supporting multispectral data, though. How about `multichannel=True, num_channels=3`, where `num_channels` is ignored when `multichannel=False`? I also would squeeze the single-channel output, to match the shape of `image_shape`. Thoughts?

Reply:
Yes, I think your proposal is better.